src/Core/Resolvers/SqlPaginationUtil.cs

```diff
@@ -520,22 +520,40 @@ public static string Base64Decode(string base64EncodedData)
         /// <summary>
         /// Create the URL that will provide for the next page of results
         /// using the same query options.
+        /// Return value formatted as a JSON array: [{"nextLink":"[base]/api/[entity]?[queryParams_URIescaped]$after=[base64encodedPaginationToken]"}]
         /// </summary>
-        /// <param name="path">The request path.</param>
-        /// <param name="queryStringParameters">Collection of query string parameters.</param>
-        /// <param name="after">The values needed for next page.</param>
-        /// <returns>The string representing nextLink.</returns>
-        public static JsonElement CreateNextLink(string path, NameValueCollection? queryStringParameters, string after)
+        /// <param name="path">The request path excluding query parameters (e.g. https://localhost/api/myEntity)</param>
+        /// <param name="queryStringParameters">Collection of query string parameters that are URI escaped.</param>
+        /// <param name="newAfterPayload">The contents to add to the $after query parameter. Should be base64 encoded pagination token.</param>
+        /// <returns>JSON element - array with nextLink.</returns>
+        public static JsonElement CreateNextLink(string path, NameValueCollection? queryStringParameters, string newAfterPayload)
         {
+            if (queryStringParameters is null)
+            {
+                queryStringParameters = new();
+            }
+            else
+            {
+                // Purge old $after value so this function can replace it.
+                queryStringParameters.Remove("$after");
+            }
+
+            // To prevent regression of current behavior, retain the call to FormatQueryString
+            // which URI escapes other query parameters. Since $after has been removed,
+            // this will not affect the base64 encoded paging token.
             string queryString = FormatQueryString(queryStringParameters: queryStringParameters);
```

Comment:
does the "after" token need to be URI escaped? or simply stay base64 encoded?

Reply:
I reviewed the discussion on [Investigate and fix use of encoding for after/nextlink · Issue #1799 · Azure/data-api-builder (github.com)](https://github.com/Azure/data-api-builder/issues/1799), where consensus was to keep value base64encoded without any URI escaping.
 
 A cursory search on google/copilot returned mixed results regarding "Should/should not URI escape that value" and no explicit RFC specification stating a value in the request body should be uri escaped.
 
While it is a URI, I'm conflicted about making an additional change to escape it in this patch.

This can be done in a different changeset if we deem it critical to change.